drivers/mcp23x17/mcp23x17.c

```diff
@@ -0,0 +1,620 @@
+/*
+ * Copyright (C) 2021 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mcp23x17
+ * @brief       Device driver implementation for Microchip MCP23x17 I/O expanders
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#include <stdint.h>
+
+#include "assert.h"
+#include "periph/i2c.h"
+#include "ztimer.h"
+
+#include "mcp23x17.h"
+#include "mcp23x17_regs.h"
+
+#if IS_USED(MODULE_MCP23X17_IRQ)
+#include "event/thread.h"
+#endif
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#if ENABLE_DEBUG
+
```

Comment:
```suggestion
#include <string.h>
```

The function `strerror` that is used in some of the DEBUG statements needs the `string.h` header.